### PR TITLE
add makefile and dockerfile

### PR DESCRIPTION
**Pull request by shahidhk**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+SETUP_MANIFEST_FILE ?= "manifests/gitkube-setup.yaml"
+
+build-controller:
+	docker build -t $(CONTROLLER_IMAGE):$(VERSION) .
+	docker push $(CONTROLLER_IMAGE):$(VERSION)
+	hack/update-image-version.sh $(SETUP_MANIFEST_FILE) $(CONTROLLER_IMAGE) $(VERSION)
+
+build-gitkubed:
+	docker build -t $(GITKUBED_IMAGE):$(VERSION) $(GITKUBED_DIR)
+	docker push $(GITKUBED_IMAGE):$(VERSION)
```
Comment (tirumaraiselvan) [inline Makefile]: should we push on build?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+# setup the working directory
+WORKDIR /go/src/github.com/hasura/gitkube
+
+# copy source code
+COPY vendor vendor
+COPY pkg pkg
+COPY cmd cmd
+
+# build the source
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /bin/gitkube cmd/gitkube-controller/main.go
```
Comment (tirumaraiselvan) [inline Dockerfile]: should we call this binary gitkube-controller?

File: Makefile (hunk 1 above)
Comment (shahidhk) [reply Makefile]: Make push a different rule?

File: Dockerfile (hunk 2 above)
Comment (shahidhk) [reply Dockerfile]: We can!

File: Makefile (hunk 1 above)
Comment (tirumaraiselvan) [reply Makefile]: Yeah, build and push should be separate because you might build many and push once.

**Pull request merged**